### ISSUE TITLE
Doesn't work on some mobile sites?

### ISSUE THREAD

**Issue by ghost** [akdor1154/fitToWidth#1]
It does not seem to anything here and on a number of other sites that have mobile versions:

http://www.theguardian.com/world/2016/may/08/north-korea-will-only-use-nuclear-weapons-if-sovereignty-is-threatened

Manual text reflow with android-text-reflow works, testing with Ffx 46.

**Comment by akdor1154** [comment]
Hi,

I intentionally disable it on sites that are designed for mobile, as they should be already designed to fit into a narrow screen!
Just to make sure I am interpreting you correctly, are you saying that normally mobile sites don't need this, but when you zoom in, you do want text to be re-flowed?

Jarrad

**Comment by ghost** [comment]
Yes, I want to zoom mobile sites and have them reflowed.  They are designed for a narrow screen but may still use too small fonts.

**Comment by akdor1154** [comment]
Out of interest, is Firefox's Text Size option an adequate fix for this?

**Comment by ghost** [comment]
Maybe I should play more with that but in the past I didn't find a setting which would work well for most sites. Much depends on other factors, when reading outside I typically want larger text because of stray light and other adverse conditions.

**Comment by akdor1154** [comment]
Cool. I am considering adding a menu item that says "Force [enable/disable] text-to-width for this domain". This preference would be remembered if you ever set it. Would that satisfy your use-case?

**Comment by ghost** [comment]
Perhaps a global option would be sufficient and easier to use and implement? I can hardly imagine a situation where I would complain if text is reflowed after zoom.

**Comment by akdor1154** [comment]
It it known to break the layout of several popular mobile sites, and the algorithm to get them to work properly would be a lot more complex (and hence unacceptably slow) that what is currently used. What about an option "Always force enable fit-to-width, THIS WILL BREAK SOME SITES"?

**Comment by ghost** [comment]
Fine for me. 

I didn't grok all details of your current implementation but is it "walking" the tree always or only after zoom events? It seems that acting only after zoom events would have advantages?

**Comment by akdor1154** [comment]
Currently it only runs on page load or when DOM nodes are added dynamically. This way zooming stays responsive (it can take a good five-ten seconds to process a complex page, which would make zooming a pain in the butt).

**Comment by ghost** [comment]
ok.. I think I understand how it works now.

**Comment by Grossdm** [comment]
@akdor1154 I have a few questions that seem to fit this issue, so I'm posting here even though it's closed. You said that the addon is disabled on mobile websites. 

- Firstly, I often find mobile websites that still need to be zoomed in
- Also, does this mean that requesting the desktop version of the site will enable your addon?
- You mentioned Firefox's text size options. If you're not aware, these are essentially gone. There is one: change text size with system setting. Tweaking about:config is not feasible browsing from one site to the next.
- You offered a per-site menu option. **To me, this sounds like an excellent feature.**

Thanks for your time and effort with this addon,
Doug®

Oh, BTW should this still be fine with new webex versions of Firefox/Fennec?

**Comment by akdor1154** [comment]
> Firstly, I often find mobile websites that still need to be zoomed in

Yep me too.

> Also, does this mean that requesting the desktop version of the site will enable your addon?

It should, the mobile switch consists of checking for a mobile-looking meta viewport tag.

> You mentioned Firefox's text size options. If you're not aware, these are essentially gone. There is one: change text size with system setting. Tweaking about:config is not feasible browsing from one site to the next.

Grr, good job Mozilla. :(

> You offered a per-site menu option. To me, this sounds like an excellent feature.

Me too but I'm pretty busy at the moment and this is a bit of work to implement. PRs welcome, or I might get to it in a few weeks. :)


In the short term I can add a "force enable on all sites" button as mentioned above.

**Comment by enomat** [comment]
> I intentionally disable it on sites that are designed for mobile, as they should be already designed to fit into a narrow screen!

There is no reason to disable it on mobile sites.

If a user want to zoom in on mobile sites he will still have to scroll because you disabled the text reflow. This defeats the purpose of this addon because the user will have the same experience he has without this addon (no text reflow).

There is no need for _"per-site menu option"_ because it is not the solution. The solution is to simly delete the code that disables this addon on mobile sites. But if you want to disable it on mobile sites by default, at least add a _"force enable on all sites"_ option.

**Comment by akdor1154** [comment]
> There is no reason to disable it on mobile sites.

> There is no need for _"per-site menu option"_ because it is not the solution. The solution is to simly delete the code that disables this addon on mobile sites.

That's just not correct sorry. This is at its core just a fancy css override and as such it does break some sites. an example at hand: github's in-comment code blocks. There will be many others.
I will still accept a pr for an opt in "force enable anywhere". N.b. I'm not actually using this add-on at the moment because I'm on Firefox Preview, I will certainly make sure it works if/when they enable arbitrary add-ons.
